src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -236,9 +236,83 @@ trait SamRecord {
   // transient attributes
   @inline final val transientAttrs: TransientAttrs = new TransientAttrs(this)
 
-  // TODO long-term: replace these two methods with methods on [[Cigar]] to save creating alignment blocks in memory
-  @inline final def refPosAtReadPos(pos: Int) = getReferencePositionAtReadPosition(pos)
-  @inline final def readPosAtRefPos(pos: Int, returnLastBaseIfDeleted: Boolean) = getReadPositionAtReferencePosition(pos, returnLastBaseIfDeleted)
```

Comment:
these are breaking changes to the API.  I like returning `Option[Int]` better, so I can return `None` instead of `0`